### DISCUSSION TITLE
AUTHENTICATION_FAILED issue

### DISCUSSION THREAD

**Discussion by mkr1d** [strongswan/strongswan#1999] [General]
Hi, 
I am using strongswan to establish a tunnel between two devices- one is a client and one is a server.
The server is CentOs7 and uses strongswan 5.9.9 with old configuration backend (ipsec.conf/secrets).
The tunnel was established successfully when the client has used Ubuntu 20.04 and strongswan 5.9.11 with new configuration backend (swanctl.conf).
The client uses TPM 2.0 Hardware device.
I am trying to replicate the same tunnel, with the exact same configuration ,this time the client uses Ubuntu 22.04 and  strongswan 5.9.11,albeit a different TPM 2.0 Hardware device (but the same handle on the TPM).
When trying to establish the connection from client side ,I get the following error:
[ENC] parsed IKE_AUTH response 1 [ N(AUTH_FAILED) ]
[IKE] received AUTHENTICATION_FAILED notify error
initiate failed: establishing CHILD_SA 'VR0_ipsec_1' failed
Browsing through the forum, I have encountered similar issue to mine: 
[https://github.com/strongswan/strongswan/discussions/1193](url)
I have inspected the logs a bit closer and found out that the 'correct' connection uses the request:
generating IKE_AUTH request 1 [ IDi CERT CERTREQ AUTH SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
while the 'wrong' connection uses the request:
generating IKE_AUTH request 1 [ IDi CERT CERTREQ AUTH SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
The IKE_AUTH requests are different,which suggests that the issue on the discussion resembles to mine.
On the discussion ,tobiasbrunner has said:
"
[tobiasbrunner](https://github.com/tobiasbrunner)
[on Aug 15, 2022](https://github.com/strongswan/strongswan/discussions/1193#discussioncomment-3398104)
Maintainer
rightid=88.xxx.xxx.xxx

This will only work if that's the identity configured on the server (which apparently it is not). To not send IDr and just check the identity against the returned certificate, you can configure rightid=%88.xxx.xxx.xxx.

Also, you want to configure the actual user name via eap_identity=FOrtner (and a corresponding EAP password in ipsec.secrets).
"
But the ipsec.conf on the server is already configured correctly:
conn %default
        keyexchange=ikev1
        authby=secret
        type=tunnel
        ike=aes128-sha1-modp2048!
        dpdaction=clear
        dpddelay=60
        dpdtimeout=180
        mobike=no

conn VR0_ipsec_1
        keyexchange=ikev2
        left=10.10.10.37
        leftsubnet=0.0.0.0/0
        right=10.10.10.38
        rightsubnet=0.0.0.0/0
        esp=aes128-sha1-modp2048!
        lifetime=8000
        ikelifetime=86400
        aggressive=no
        **rightid=%any**
        type=tunnel
        lifebytes=4608000000
        authby=rsasig
        leftcert=/etc/ipsec.d/certs/VR0_ipsec_1.cert
        auto=route
        mark=1

So,my question is how do I either make the request in 'correct' form or configure the ipsec.conf to be able to authorize without issues?

**Comment by tobiasbrunner** [comment]
Why do you think this has anything to do with that discussion? Just because it's an AUTH_FAILED notify? That can have a multitude of different causes. Which is why you should start by reading the log of the host that actually produced that error (i.e. the server) to see why it did so.

By the way, the only difference between the two requests visible in those logs is the additional address sent for MOBIKE. That provides absolutely no indication as to what the actual problem is.
